sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1169,3 +1170,31 @@ object ComputeCurrentTime extends Rule[LogicalPlan] {
     }
   }
 }
+
+/**
+ * Replace the `UpCast` expression by `Cast`, and throw exceptions if the cast may truncate.
+ */
+object ResolveUpCast extends Rule[LogicalPlan] {
+  private def fail(from: Expression, to: DataType) = {
+    throw new AnalysisException(s"Cannot up cast `${from.prettyString}` from " +
```

Comment:
Here I just use `prettyString` to show the field path like `a`, `a.b`, should we also show the type path like we did before?